haystack/document_stores/opensearch.py

```diff
@@ -438,6 +491,18 @@ def query_by_embedding(
         if return_embedding is None:
             return_embedding = self.return_embedding
 
+        if self.index_type in ["ivf", "ivf_pq"] and not self._ivf_model_exists(index=index):
+            add_num_of_embs = ""
+            if self.ivf_train_size != 0:
+                embs_to_add = self.get_embedding_count(index=index, headers=headers) - self.ivf_train_size
+                add_num_of_embs = (
+                    f"or add at least {embs_to_add} more embeddings to automatically start the " f"training process "
+                )
+            raise DocumentStoreError(
+                f"Index of type '{self.index_type}' is not trained yet. Train the index manually using "
+                f"`train_index` {add_num_of_embs}before querying it."
+            )
+
```

Comment:
Nice! Can we please add a test case for this.

Reply:
And we need the same for `query_by_embedding_batch`. It's implemented in `search_engine.py` however. You might also consider moveing this check into `_construct_dense_query_body`.